hub-saml/src/test/java/uk/gov/ida/saml/hub/transformers/outbound/HubEidasAttributeQueryRequestToSamlAttributeQueryTransformerTest.java

```diff
@@ -0,0 +1,212 @@
+
+    @Mock
+    private AttributeQueryAttributeFactory attributeQueryAttributeFactory;
+
+    @Mock
+    private EncryptedAssertionUnmarshaller encryptedAssertionUnmarshaller;
+
+    @Before
+    public void setup() throws Exception {
+        openSamlXmlObjectFactory = new OpenSamlXmlObjectFactory();
+        HubAssertionMarshaller assertionTransformer = new HubAssertionMarshaller(openSamlXmlObjectFactory, attributeFactory, outboundAssertionToSubjectTransformer);
+        AssertionFromIdpToAssertionTransformer assertionFromIdpAssertionTransformer = new AssertionFromIdpToAssertionTransformer(stringAssertionTransformer);
+
+        when(stringAssertionTransformer.apply(anyString())).thenReturn(anAssertion().buildUnencrypted());
+
+        transformer = new HubEidasAttributeQueryRequestToSamlAttributeQueryTransformer(
+            openSamlXmlObjectFactory,
+            assertionTransformer,
+            assertionFromIdpAssertionTransformer,
+            attributeQueryAttributeFactory,
+            encryptedAssertionUnmarshaller);
+    }
+
+    @Test
+    public void shouldTransformProperly() throws Exception {
+        PersistentId persistentId = aPersistentId().build();
+        HubEidasAttributeQueryRequest hubEidasAttributeQueryRequest = aHubEidasAttributeQueryRequest()
+            .withId("originalId")
+            .withPersistentId(persistentId)
+            .build();
+
+        AttributeQuery transformedQuery = transformer.apply(hubEidasAttributeQueryRequest);
+
+        assertThat(transformedQuery.getID()).isEqualTo(hubEidasAttributeQueryRequest.getId());
+        assertThat(transformedQuery.getSubject().getNameID().getValue()).isEqualTo(persistentId.getNameId());
+        assertThat(transformedQuery.getIssuer().getValue()).isEqualTo(hubEidasAttributeQueryRequest.getIssuer());
+        assertThat(transformedQuery.getVersion()).isEqualTo(SAMLVersion.VERSION_20);
+        new XmlObjectToBase64EncodedStringTransformer<>().apply(transformedQuery);
+    }
+
+    @Test
+    public void shouldIncludeCycle3Assertion() throws Exception {
+        final HubAssertion cycle3DataAssertion = aHubAssertion().build();
+
+        HubEidasAttributeQueryRequest hubEidasAttributeQueryRequest = aHubEidasAttributeQueryRequest()
+            .withCycle3DataAssertion(cycle3DataAssertion)
+            .build();
+
+        AttributeQuery transformedQuery = transformer.apply(hubEidasAttributeQueryRequest);
+
+        List<XMLObject> unknownXMLObjects = transformedQuery.getSubject().getSubjectConfirmations().get(0).getSubjectConfirmationData().getUnknownXMLObjects(Assertion.DEFAULT_ELEMENT_NAME);
+        assertThat(unknownXMLObjects.size()).isEqualTo(1);
+        assertThat(((Assertion) unknownXMLObjects.get(0)).getID()).isEqualTo(cycle3DataAssertion.getId());
+    }
+
+    @Test
+    public void shouldIncludeEncryptedIdentityAssertionOnly() throws Exception {
+        final String encryptedIdentityAssertion = ENCRYPTED_IDENTITY_ASSERTION;
```

Comment:
Why make a final copy of a constant?

Reply:
Good spot. I used refactor to extract this variable into constant field. I have removed it.